bgpd/bgp_attr.c

```diff
@@ -1073,13 +1073,13 @@ void bgp_attr_undup(struct attr *new, struct attr *old)
 	if (new->aspath != old->aspath)
 		aspath_free(new->aspath);
 
-	if (new->community != old->community)
+	if (new->community && new->community != old->community)
```

Comment:
wouldn't be better doing this in `community_free()`? I mean just:
```
if (!(*com))
  return;
```
Because I see we have more places where this isn't checked.

Reply:
Yep, this is the correct patch.

@joshdcox do you mind updating your PR to do that instead?

Reply:
Probably. I was trying to only impact code I know is broken, but your approach sounds better. Should I commit that change to this PR or open a new one?

Reply:
Use the same commit, just append and use `git commit --amend` :+1: